app/controllers/api/vcs.php

```diff
@@ -0,0 +1,1019 @@
+        if ($cursor) {
+            /** @var Query $cursor */
+            $vcsInstallationId = $cursor->getValue();
+            $cursorDocument = $dbForConsole->getDocument('vcsInstallations', $vcsInstallationId);
+
+            if ($cursorDocument->isEmpty()) {
+                throw new Exception(Exception::GENERAL_CURSOR_NOT_FOUND, "Installation '{$vcsInstallationId}' for the 'cursor' value not found.");
+            }
+
+            $cursor->setValue($cursorDocument);
+        }
+
+        $filterQueries = Query::groupByType($queries)['filters'];
+
+        $results = $dbForConsole->find('vcsInstallations', $queries);
+        $total = $dbForConsole->count('vcsInstallations', $filterQueries, APP_LIMIT_COUNT);
+
+        if (\count($results) > 0) {
+            $installationIds = \array_map(fn ($result) => $result->getInternalId(), $results);
+
+            $functions = Authorization::skip(fn () => $dbForProject->find('functions', [
+                Query::equal('vcsInstallationInternalId', \array_unique($installationIds)),
+                Query::limit(APP_LIMIT_SUBQUERY)
+            ]));
+
+            foreach ($results as $result) {
+                $installationFunctions = \array_filter($functions, fn ($function) => $function->getAttribute('vcsInstallationInternalId') === $result->getInternalId());
+
+                $result->setAttribute('functions', $installationFunctions);
+            }
+        }
+
+        $response->dynamic(new Document([
+            'installations' => $results,
+            'total' => $total,
+        ]), Response::MODEL_INSTALLATION_LIST);
+    });
+
+App::get('/v1/vcs/installations/:installationId')
+    ->groups(['api', 'vcs'])
+    ->desc('Get installation')
+    ->label('scope', 'public')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'vcs')
+    ->label('sdk.method', 'getInstallation')
+    ->label('sdk.description', '/docs/references/vcs/get-installation.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_INSTALLATION)
+    ->param('installationId', '', new Text(256), 'Installation Id')
+    ->inject('response')
+    ->inject('project')
+    ->inject('dbForProject')
+    ->inject('dbForConsole')
+    ->action(function (string $installationId, Response $response, Document $project, Database $dbForProject, Database $dbForConsole) {
+        $installation = $dbForConsole->getDocument('vcsInstallations', $installationId);
+
+        if ($installation === false || $installation->isEmpty()) {
+            throw new Exception(Exception::INSTALLATION_NOT_FOUND);
+        }
+
+        if ($installation->getAttribute('projectInternalId') !== $project->getInternalId()) {
+            throw new Exception(Exception::INSTALLATION_NOT_FOUND);
+        }
+
+        $functions = Authorization::skip(fn () => $dbForProject->find('functions', [
```

Comment:
Do we have to link functions to the installation like this? Where do we use it?
This can become very problematic once we have many functions. If this is just for the UI, we can find better ways for it to get this information.